src/main/groovy/org/shipkit/internal/gradle/GitPlugin.java

```diff
@@ -111,9 +113,16 @@ public void execute(final Task t) {
 
         TaskMaker.execTask(project, COMMIT_CLEANUP_TASK, new Action<Exec>() {
             public void execute(final Exec t) {
-                t.setDescription("Removes last commit, using 'reset --hard HEAD~'");
-                //TODO replace with combination of 'git reset --soft HEAD~ && git stash' so that we don't lose commits
-                t.commandLine("git", "reset", "--hard", "HEAD~");
+                t.setDescription("Stashes current changes");
+                t.commandLine("git", "stash");
+                t.dependsOn(RESET_COMMIT_TASK);
+            }
+        });
+
+        TaskMaker.execTask(project, RESET_COMMIT_TASK, new Action<Exec>() {
+            public void execute(final Exec t) {
```

Comment:
Thx for your feedback! Yes, indeed, we'll end up in a cleaner solution after applying your suggestions. I will update the PR soon.